src/inner_product_proof.rs

```diff
@@ -214,11 +227,84 @@ impl InnerProductProof {
                 .chain(self.R_vec.iter()),
         );
 
-        if expect_P == *P {
-            Ok(())
-        } else {
-            Err(())
+        if expect_P == *P { Ok(()) } else { Err(()) }
+    }
+
+    /// Returns the size in bytes required to serialize the inner product proof.
+    /// For `n` multiplications the proof size is \\(32 \cdot (2\lg n+2)\\) bytes.
+    pub fn serialized_size(&self) -> usize {
+        (self.L_vec.len() * 2 + 2) * 32
+    }
+
+    /// Serializes the proof into a byte array of \\(2n+2\\) 32-byte elements.
+    /// The layout of the inner product proof is:
+    /// * \\(n\\) compressed Ristretto points \\(L_0, \dots, L_{n-1}\\),
+    /// * \\(n\\) compressed Ristretto points \\(R_0, \dots, R_{n-1}\\),
+    /// * two scalars \\(a, b\\).
```

Comment:
Does it make sense to interleave the L,R points, since that's the order in which they're generated and fed into the transcript?

Reply:
Yeah, that might be cleaner. Will try it.